### PR TITLE
docs: Add group field context

### LINKED ISSUE #8562: ignoreDifferences default group undocumented/leaving empty does not work on Ingress
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

I have an ingress resource which I would like to ignore differences on certain json path fields, e.g. /spec/rules or /metadata/annotations, for the usual reasons we kubernetes people do not want to talk about, dirty hacks, special use cases, this and that, in this case we have some mechanism which temporarily changes backend service.

This ingress resource comes from a helm chart, though I do not believe this would be important to the issue.
I can not include the whole application YAML file, but here the part of the spec section I had to work with:

```
spec: 
  project: default

  destination:
    server: 'https://kubernetes.default.svc'
    namespace: some-namespace
  syncPolicy:
    automated:
      selfHeal: true
    syncOptions:
    - RespectIgnoreDifferences=true

  ignoreDifferences:
  - kind: "Ingress"
    jsonPointers:
    - /spec/rules
    - /metadata/annotations
 ```

Specifying it like this, does not lead to ignoring the annotations or rule changes; all changes made e.g. via `kubectl edit` or the editor in the webinterface or some kubectl are immediately reverted.

In the in-cluster `applications.argoproj.io` resources I do not see anything unexpected, so there is not an accidental `group: apps` directive (or leftover from first attempts using too much copypasta) or something like that.

Adding `group: "networking.k8s.io"` fixes my problem though, resulting in:
```
spec: 
  project: default

  destination:
    server: 'https://kubernetes.default.svc'
    namespace: some-namespace
  syncPolicy:
    automated:
      selfHeal: true
    syncOptions:
    - RespectIgnoreDifferences=true

  ignoreDifferences:
  - kind: "Ingress"
    group: "networking.k8s.io"
    jsonPointers:
    - /spec/rules
    - /metadata/annotations
```

**To Reproduce**

See above. Have an Ingress resource and try to ignore parts of it, without specifying the group explicitely. Possibly not-works with other resources not in the apps group as well, I have not tested this with other kinds.

**Expected behavior**

Documented behaviour of the `group` directive and its defaults, or leaving out the group object should default to match all resources with the correct kind, independent of group. For me I would totally be fine with some update to the documentation describing how it works, but as the group also did not 

All the examples and documentation use the `apps` group without further specifying what this refers to. I did find the correct group by inspecting the in-cluster `applications.argoproj.io` resource.

**Version**

Installed on k8s via manifests/install.yaml from release-2.3 branch.

```shell
argocd: v2.3.0-rc5+aa54aa7
  BuildDate: 2022-02-04T23:39:21Z
  GitCommit: aa54aa7eda70f5e049c9e25df75221d899efff4b
  GitTreeState: clean
  GoVersion: go1.17.6
  Compiler: gc
  Platform: linux/amd64
FATA[0000] Argo CD server address unspecified   
```

**Logs**

I would not know what kind of logs would be useful here.

### PR DESCRIPTION
**Pull request by leoluz**
Signed-off-by: Leonardo Luz Almeida <leonardo_almeida@intuit.com>

Fixes https://github.com/argoproj/argo-cd/issues/8562

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: docs/user-guide/diffing.md [hunk 1]
```diff
@@ -32,6 +32,7 @@ spec:
     - /spec/replicas
 ```
 
+Note that the `group` field relates to the kubernetes API group.
```
Comment (crenshaw-dev) [inline docs/user-guide/diffing.md:35]:
```suggestion
Note that the `group` field relates to the [Kubernetes API group](https://kubernetes.io/docs/reference/using-api/#api-groups).
```

Maybe helpful?

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]

File: docs/user-guide/diffing.md (hunk 1 above)
Comment (jannfis) [reply docs/user-guide/diffing.md:35]: Plus it probably is helpful to note that it must not include the API version, only the group. I've seen a couple of times that people will specify `group` as `apps/v1` and similar.

**Comment by leoluz** [comment]
Comments addressed.
Pls take a look @crenshaw-dev @jannfis

**Review by crenshaw-dev — APPROVED** [review]

**Review by jannfis — APPROVED** [review]
LGTM
Thanks @leoluz !

**Pull request merged**